apps/ios/SimpleXChat/ErrorAlert.swift

```diff
@@ -47,25 +70,39 @@ extension LocalizedStringKey: @unchecked Sendable { }
 
 extension View {
     /// Bridges ``ErrorAlert`` to the generic alert API.
+    /// - Parameters:
+    ///   - errorAlert: Binding to the Error, which is rendered in the alert
+    ///   - actions: View Builder containing action buttons.
+    ///   Will default to `Ok` dismiss error action, when not provided.
+    ///   Will implicitly add `Cancel` action, if a destructive action is present
+    ///
+    /// - Returns: View, which displays ErrorAlert?, when set.
     @ViewBuilder public func alert<A: View>(
-        _ errorAlert: ErrorAlert?,
-        @ViewBuilder actions: (ErrorAlert) -> A
+        _ errorAlert: Binding<ErrorAlert?>,
+        @ViewBuilder actions: (ErrorAlert) -> A = { _ in EmptyView() }
     ) -> some View {
-        if let alert = errorAlert {
-            if let message = alert.message {
-                self.alert(
-                    alert.title,
-                    isPresented: .constant(errorAlert != nil),
-                    actions: { actions(alert) },
-                    message: { Text(message) }
-                )
-            } else {
-                self.alert(
-                    alert.title,
-                    isPresented: .constant(errorAlert != nil),
-                    actions: { actions(alert) }
-                )
-            }
+        if let alert = errorAlert.wrappedValue {
+            self.alert(
+                alert.title,
+                isPresented: Binding<Bool>(
+                    get: { errorAlert.wrappedValue != nil },
+                    set: { if !$0 { errorAlert.wrappedValue = nil } }
+                ),
+                actions: {
+                    if let actions = alert.actionsOverride {
+                        ForEach(actions) {
+                            Button($0.titleKey, role: $0.role, action: $0.action ?? { })
+                        }
+                    } else { actions(alert) }
+                },
+                message: {
+                    if let message = alert.message {
+                        Text(message)
+                    } else {
+                        EmptyView()
+                    }
+                }
+            )
```

Comment:
Tested and SwiftUI will work fine, with `EmptyView` being passed - so no needed to use different initialisers.